packages/jsdoc-plugins/lib/relation-fixer/addmissingdoclets.js

```diff
@@ -67,36 +85,41 @@ function addMissingDoclets( originalDoclets ) {
 		}
 	];
 
-	for ( const childDoclet of entitiesWhichNeedNewDoclets ) {
-		for ( const setting of settings ) {
+	for ( const extensibleDoclet of extensibleDoclets ) {
+		for ( const option of options ) {
 			const missingDocletsData = getMissingDocletsData(
 				docletCollection,
-				childDoclet,
-				setting
+				extensibleDoclet,
+				option
 			);
 
 			newDocletsToAdd.push( ...missingDocletsData.newDoclets );
 			docletsToIgnore.push( ...missingDocletsData.docletsWhichShouldBeIgnored );
 		}
 	}
 
-	docletsToIgnore.forEach( d => {
-		d.ignore = true;
-	} );
-	clonedDoclets.push( ...newDocletsToAdd );
-
-	extendTypedefs( typedefDoclets );
+	// Ignore doclets that shouldn't be used anymore. They will be removed afterward.
+	for ( const docletToIgnore of docletsToIgnore ) {
+		docletToIgnore.ignore = true;
+	}
 
-	return clonedDoclets;
+	return [
+		...doclets,
+		...newDocletsToAdd
+	];
 }
 
-// Copy properties from parent typedefs to typedefs which extend them.
+/**
+ * Copy properties from parent typedefs to typedefs which extend them.
+ *
+ * @param {Array.<Doclet>} typedefDoclets
+ */
 function extendTypedefs( typedefDoclets ) {
```

Comment:
As I said https://github.com/ckeditor/ckeditor5-dev/pull/483/files#r256765868 – this function should return a new object instead of modifying passed parameters. Even if it was done earlier, now we are changing this part of code and this bug should be fixed if it isn't too hard.